specs/Print.md

```diff
@@ -0,0 +1,816 @@
+  COREWEBVIEW2_PRINT_COLOR_MODE_COLOR,
+
+  /// Indicate that the printed output will be in shades of gray.
+  COREWEBVIEW2_PRINT_COLOR_MODE_GRAYSCALE,
+
+} COREWEBVIEW2_PRINT_COLOR_MODE;
+
+/// Specifies the collation for a print.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_COLLATION {
+  /// The default collation for a printer.
+  COREWEBVIEW2_PRINT_COLLATION_DEFAULT,
+
+  /// Indicate that the collation has been selected for the printed output.
+  COREWEBVIEW2_PRINT_COLLATION_COLLATED,
+
+  /// Indicate that the collation has not been selected for the printed output.
+  COREWEBVIEW2_PRINT_COLLATION_UNCOLLATED,
+
+} COREWEBVIEW2_PRINT_COLLATION;
+
+/// Specifies the media size for a print.
+[v1_enum] typedef enum COREWEBVIEW2_PRINT_MEDIA_SIZE {
+  /// The default media size for a printer.
+  COREWEBVIEW2_PRINT_MEDIA_SIZE_DEFAULT,
+
+  /// Indicate custom media size that is specific to the printer.
+  COREWEBVIEW2_PRINT_MEDIA_SIZE_CUSTOM,
+
+} COREWEBVIEW2_PRINT_MEDIA_SIZE;
+
+[uuid(92269C88-1CC1-47E0-9A87-71458409BA15), object, pointer_default(unique)]
+interface ICoreWebView2_15 : ICoreWebView2_14 {
+  /// Print the current web page asynchronously to the specified printer with the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings.
+  /// The handler will return `errorCode` as `S_OK` and `isSuccessful` as FALSE if `printerName`
+  /// doesn't match with the name of any installed printers on the user OS.
+  /// The handler will return `errorCode` as `E_INVALIDARG` and `isSuccessful` as FALSE
+  /// if the caller provides invalid settings for a given printer.
+  ///
+  /// The async `Print` operation completes when it finishes printing to the printer.
+  /// At this time the `ICoreWebView2PrintCompletedHandler` is invoked. Only one `Printing` operation
+  /// can be in progress at a time. If `Print` is called while a `Print` or `PrintToPdf`
+  /// or `PrintToPdfStream` or `ShowPrintUI` job is in progress, the completed handler is
+  /// immediately invoked with `E_ABORT` and `isSuccessful` set to FALSE.
+  ///
+  /// |       errorCode     |       isSuccessful    |               Notes                                                                           |
+  /// | --- | --- | --- |
+  /// |        S_OK         |        True           | Print operation succeeded.                                                                    |
+  /// |        S_OK         |        False          | If specified printer is not found or printer status is not available, offline or error state. |
+  /// |       E_FAIL        |        False          | Print operation is failed.                                                                    |
+  /// |     E_INVALIDARG    |        False          | If the caller provides invalid settings for the specified printer.                            |
+  /// |       E_ABORT       |        False          | Print operation is failed as printing job already in progress.
```

Comment:
What about unsupported settings? Suppose I ask for duplex printing from a printer that doesn't support it.

Is the duplex request ignored, and I get simplex printing? Or does this produce some error like `E_NOT_SUPPORTED`

Reply:
We are not resulting failure for these cases, we ignore the value and simplex is printed.

Reply:
We need to be clear about what "invalid settings" means. It seems that it's largely arbitrary whether asking for something that the printer can't do counts as "invalid" or "not supported, ignored".

Reply:
discussion: Invalid error is something that never makes sense for any printer versus something that is invalid for that specific printer.
It will be good to update the documentation to clarify this.
Also good to document that job already in progress is only for other calls to PrintAsync on the one WebView2
Please also validate that it shouldn't collide for ShowPrintUI calls only programmatic calls to PrintAsync/PrintToPdfStreamAsync.
Investigate if its possible to relax requirement of can't have multiple prints at one time.

Reply:
Updated the documentation accordingly. Looks like underlying code not supporting printing when a printing for render frame is in progress.